core/src/main/scala/kafka/controller/ControllerContext.scala

```diff
@@ -295,6 +316,10 @@ class ControllerContext {
     topicsToBeDeleted -= topic
     topicsWithDeletionStarted -= topic
     allTopics -= topic
+    if (topicIds.get(topic)!= None) {
+      topicNames.remove(topicIds.get(topic).get)
+      topicIds.remove(topic)
+    }
```

Comment:
We could do:
```
topicIds.remove(topic).foreach { topicId =>
  topicNames.remove(topicId)
}
```

Reply:
I was thinking about this some more, and I don't think we will have a case where the topic ID is not in the map (the if conditional is not needed.) However, the suggestion you gave is still good because it prevents the extra lookup.